api/src/main/java/io/camunda/zeebe/process/test/api/InMemoryEngine.java

```diff
@@ -29,10 +30,15 @@
    */
   void increaseTime(Duration timeToAdd);
 
+  /** Waits for the engine to reach an idle state. */
+  void waitForIdleState();
+
   /**
-   * Waits for the engine to reach an idle state.
+   * Waits for the engine to start processing.
    *
-   * <p>For more info on the idle state refer to {@link IdleStateMonitor}
+   * @param timeout the maximum amount of time to wait before processing state has been reached
+   * @throws InterruptedException if the current thread was interrupted while waiting
+   * @throws TimeoutException if the engine has not reached a processing state before the timeout
    */
-  void waitForIdleState();
+  void waitForProcessingState(Duration timeout) throws InterruptedException, TimeoutException;
```

Comment:
:wrench: 
```suggestion
  void waitForBusyState(Duration timeout) throws InterruptedException, TimeoutException;
```

Maybe also change order of methods:
1. Wait for Busy
2. Wait for Idle

Reply:
:wrench: Maybe harmonize interfaces and also add a timeout to `waitForIdleState`

Reply:
btw, totally fine to keep the name processing state in the idle state monitor, because there it is more technical and closer to the implementation. But here I have a slight preference for using the busy idle pair. But it's only a mild preference, so I am also ok with sticking with the name.